datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -521,18 +554,111 @@ impl FileOpener for ParquetOpener {
     }
 }
 
-// Check PruningPredicates just work on one column.
-fn check_page_index_push_down_valid(predicate: &Option<PruningPredicate>) -> bool {
-    if let Some(predicate) = predicate {
-        // for now we only support pushDown on one col, because each col may have different page numbers, its hard to get
-        // `num_containers` from `PruningStatistics`.
-        let cols = predicate.need_input_columns_ids();
-        //Todo more specific rules
-        if cols.len() == 1 {
-            return true;
+// combine two `RowSelection` return the intersection
+// For example:
+// self:     NNYYYYNNY
+// other:    NYNNNNNNY
+//
+// returned: NNNNNNNNY
+// set `need_combine` true will combine result: Select(2) + Select(1) + Skip(2) -> Select(3) + Skip(2)
+pub(crate) fn intersect_row_selection(
```

Comment:
Will move this to `arrow-rs`

Reply:
Filed https://github.com/apache/arrow-rs/issues/3003